packages/backend-function/src/function_env_translator.ts

```diff
@@ -87,7 +87,7 @@ export class FunctionEnvironmentTranslator {
     this.lambda.node.addValidation({
       validate: (): string[] => {
         new FunctionEnvironmentTypeGenerator(
-          this.lambda.node.id,
+          this.lambda.node.id.replace(/-lambda$/, ''), // Remove -lambda suffix from node id and keep the lambda name
```

Comment:
rather than trying to derive this from the node id, should we just plumb the function name from the factory down to this location?

Reply:
I think the "most correct" refactor here would be to inject a FunctionEnvironmentTypeGenerator into the FunctionEnvironmentTranslator ctor. Then change `generateTypedProcessEnvShim()` to take in the list of env vars rather than storing it as a class member.

I'll leave it to you if you want to do that refactor. What you've got here isn't a big deal and it's all internal so can be refactored any time.

Reply:
Did the refactor as the longer I looked at that `replace` the more I felt it needed to be done lol